src/FixedDataTable.js

```diff
@@ -964,12 +964,29 @@ var FixedDataTable = createReactClass({
       children.push(child);
     });
 
-    // Allow room for the scrollbar, less 1px for the last column's border
-    var adjustedWidth = props.width - Scrollbar.SIZE - Scrollbar.OFFSET;
-
+    // Figure out if the vertical scrollbar will be visible first, 
+    // because it will determine the width of the table
     var useGroupHeader = false;
+    var groupHeaderHeight = 0;
+
     if (children.length && children[0].type.__TableColumnGroup__) {
       useGroupHeader = true;
+      groupHeaderHeight = props.groupHeaderHeight;
+    }
+
+    var useMaxHeight = props.height === undefined;
+    var height = Math.round(useMaxHeight ? props.maxHeight : props.height);
+    var totalHeightReserved = props.footerHeight + props.headerHeight +
+        groupHeaderHeight + 2 * BORDER_HEIGHT;
+    var bodyHeight = height - totalHeightReserved;
```

Comment:
This is very close, but I think may have an issue where totalHeightReserved here doesn't take into account the horizontal scrollbar if one is needed.

Do you want to pair program through the last mile?

Reply:
Ah I see now.  Sweet.
I think you can remove the maxScrollY line on 1030 since it's overridden below on 1195.

Is there a risk that doing the adjustedWidth changes on 1197 could mess up the flex width computations on 1109 & 1116?  Or will that only happen if we decided we needed a horizontal scrollbar late enough that we're not expanding the flex columns?

If that last point checks out, we can merge & I'll cut a release

Reply:
After looking into this I don't believe it will be an issue because as you said the 2nd width adjustment only occurs if a horizontal scrollbar appears late, and then only if we haven't already taken the offset into account with the 1st adjustment.. Because the flex width [only needs to be calculated](https://github.com/schrodinger/fixed-data-table-2/blob/master/src/FixedDataTableWidthHelper.js#L141-L143) when the current column width is less than the table width (aka, no horizontal scrollbar visible), we are safe. Essentially, these two scenarios are mutually exclusive.